frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.tsx

```diff
@@ -628,6 +644,24 @@ class DashboardHeaderContainer extends Component<DashboardHeaderProps> {
   }
 }
 
+function getDisabledSaveButtonTooltip(
+  missingRequiredParams: UiParameter[],
+): string {
+  if (!missingRequiredParams.length) {
+    return "";
+  }
+
+  const names = missingRequiredParams
+    .map(param => `"${param.name}"`)
+    .join(", ");
+
+  return ngettext(
+    msgid`The ${names} parameter requires a default value but none was provided.`,
+    `The ${names} parameters require default values but none were provided.`,
+    missingRequiredParams.length,
```

Comment:
nit: messages explicitly depend on `names`, and only implicitly on `missingRequiredParams`

```suggestion
    names.length,
```

Reply:
This is not right because `names` is a concatenated string of the shape "param1, param2". What's funny is that I actually did that myself previously and then fixed that as a bug, because of course, the length of the string is always > 0.